planner/core/plan_cache_lru.go

```diff
@@ -0,0 +1,190 @@
+// Package core Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package core
+
+import (
+	"container/list"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/util/kvcache"
+)
+
+// CacheEntry wraps Key and Value. It's the value of list.Element.
+type CacheEntry struct {
+	PlanKey   kvcache.Key
+	PlanValue kvcache.Value
+}
+
+// LRUPlanCache is a simple least recently used cache, not thread-safe, JUST use for plan cache.
```

Comment:
For safety, please make it thread-safe. I suggest that we can simply add a `sync.Mutex` to make it thread-safe.